backend/nfconfig/config.go

```diff
@@ -124,11 +130,67 @@ func sortPlmnSnssaiConfig(plmnSnssai []nfConfigApi.PlmnSnssai) {
 	}
 }
 
-func (c *inMemoryConfig) syncAccessAndMobility() {
-	c.accessAndMobility = []nfConfigApi.AccessAndMobility{}
+func (c *inMemoryConfig) syncAccessAndMobility(networkSlices []configmodels.Slice) {
+	plmnSnssaiTacsMap := map[accessAndMobilityKey][]string{}
+	for _, s := range networkSlices {
+		accessAndMobilityTmp := accessAndMobilityKey{
+			plmn:    s.SiteInfo.Plmn,
+			sliceId: s.SliceId,
+		}
+		if plmnSnssaiTacsMap[accessAndMobilityTmp] != nil {
+			logger.NfConfigLog.Warnf("Found duplicate Network slice `%+v` for PLMN `%+v`, merging TACs for Access and Mobility", s.SliceId, s.SiteInfo.Plmn)
+		} else {
+			plmnSnssaiTacsMap[accessAndMobilityTmp] = []string{}
+		}
+		for _, g := range s.SiteInfo.GNodeBs {
+			tac := strconv.Itoa(int(g.Tac))
+			if !slices.Contains(plmnSnssaiTacsMap[accessAndMobilityTmp], tac) {
+				plmnSnssaiTacsMap[accessAndMobilityTmp] = append(plmnSnssaiTacsMap[accessAndMobilityTmp], tac)
+			}
+		}
+	}
+	c.accessAndMobility = convertPlmnSnssaiTacsMapToSortedList(plmnSnssaiTacsMap)
 	logger.NfConfigLog.Debugln("Updated Access and Mobility in-memory configuration. New configuration:", c.accessAndMobility)
 }
 
+func convertPlmnSnssaiTacsMapToSortedList(plmnSnssaiMap map[accessAndMobilityKey][]string) []nfConfigApi.AccessAndMobility {
+	newAccessAndMobilityConfig := []nfConfigApi.AccessAndMobility{}
+	for plmnSliceId, tacList := range plmnSnssaiMap {
+		plmnId := nfConfigApi.NewPlmnId(plmnSliceId.plmn.Mcc, plmnSliceId.plmn.Mnc)
+		parsedSnssai, err := parseSnssaiFromSlice(plmnSliceId.sliceId)
+		if err != nil {
+			logger.NfConfigLog.Warnf("Error in parsing SNSSAI: %v. Network slice `%+v` will be ignored", err, plmnSliceId.sliceId)
+			continue
+		}
+		accessAndMobility := nfConfigApi.NewAccessAndMobility(*plmnId, parsedSnssai)
+		accessAndMobility.Tacs = tacList
```

Comment:
Before adding taclist we can simply sort it. By this way we can remove following code:
```
	for i := range accessAndMobility {
		slices.Sort(accessAndMobility[i].Tacs)
	}
```
```suggestion
				tacList := make([]string, 0, len(tacSet))
		for tac := range tacSet {
			tacList = append(tacList, tac)
		}
		slices.Sort(tacList)
```

Reply:
I am not sure about the increase in efficiency here. Also, I would prefer to maintain all the sorting logic in the same function (`sortAccessAndMobilityConfig`).

Reply:
OK, that's fine.